### PR TITLE
Add SYCL logic in whisper

### PR DESCRIPTION
**Pull request by abhilash1910**
Following abstraction from #2690 [Feature: Integrate with unified SYCL backend for Intel GPUs](https://github.com/ggerganov/llama.cpp/pull/2690)  some modifications are made to ensure functionality with SYCL on whisper. The ggml sycl files are already updated and was missing some frontend and CI pipelines. 
@airMeng @luoyu-intel  @NeoZhangJianyu  please review once when available. (Keeping it as draft as examples need to be updated) 
(cc @ggerganov )

### CODE REVIEW

**Comment by abhilash1910** [comment]
@ggerganov  the android build https://github.com/ggerganov/whisper.cpp/actions/runs/8001183554/job/21851909961 seems to be failing . Requesting review .

**Comment by ggerganov** [comment]
I'm hoping that after the latest changes, the CI will be green and then will review this PR

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required (VERSION 3.5)
 
 project(whisper.cpp VERSION 1.5.4)
-
+include(CheckIncludeFileCXX)
```
Comment (ggerganov) [inline CMakeLists.txt:4]: Is this used?

Comment (abhilash1910) [reply CMakeLists.txt:4]: yes this is not required. will update.

**Review by ggerganov — APPROVED** [review]

**Comment by hlevring** [comment]
@abhilash1910 are you still working on this to support windows?

**Pull request merged**